spring-cloud-gcp-pubsub/src/main/java/com/google/cloud/spring/pubsub/support/DefaultSubscriberFactory.java

```diff
@@ -305,21 +311,89 @@ public SubscriberStub createSubscriberStub(String subscriptionName) {
 		}
 	}
 
-	SubscriberStubSettings buildSubscriberStubSettings(String subscriptionName) throws IOException {
-		SubscriberStubSettings.Builder subscriberStubSettings = SubscriberStubSettings.newBuilder();
+	SubscriberStubSettings buildGlobalSubscriberStubSettings() throws IOException {
+		SubscriberStubSettings.Builder subscriberStubSettings = buildStubSettingsWithoutConfigurations();
 
-		if (this.credentialsProvider != null) {
-			subscriberStubSettings.setCredentialsProvider(this.credentialsProvider);
+		if (this.pullEndpoint != null) {
+			subscriberStubSettings.setEndpoint(this.pullEndpoint);
+		}
+		else {
+			setGlobalPullEndpoint(subscriberStubSettings);
+		}
+
+		ExecutorProvider executor = this.executorProvider != null ? this.executorProvider
+				: createGlobalExecutorProvider();
+		if (executor != null) {
+			subscriberStubSettings.setBackgroundExecutorProvider(executor);
 		}
 
+		if (this.subscriberStubRetrySettings != null) {
+			subscriberStubSettings.pullSettings().setRetrySettings(this.subscriberStubRetrySettings);
+		}
+		else {
+			setGlobalRetrySettings(subscriberStubSettings);
+		}
+
+		if (this.retryableCodes != null) {
+			subscriberStubSettings.pullSettings().setRetryableCodes(
+					this.retryableCodes);
+		}
+
+		return subscriberStubSettings.build();
+	}
+
+	private void setGlobalPullEndpoint(SubscriberStubSettings.Builder subscriberStubSettings) {
+		if (this.pubSubConfiguration == null) {
```

Comment:
Would it be possible to avoid allowing `pubSubConfiguraiton` to be `null`?
Can we just create an empty `PubSubConfiguration` in the deprecated constructor?
Also, maybe we can remove a lot of the fields in `DefaultSubscriberFactory` and rely on the `pubSubConfiguration` instead? For example, you can then have:
```
	public void setParallelPullCount(Integer parallelPullCount) {
		this.pubSubConfiguration.getSubscriber().setParallelPullCount(parallelPullCount);
	}
```
This way you may be able to remove a lot of the boilerplate code.

Reply:
Not setting pubSubConfiguration to `null` makes sense! However, I'm a little hesitant on moving the fields out of `DefaultSubscriberFactory` at the moment since the setters for these fields are public. Additionally, some of these configurations will have to be registered as beans so that might help reduce the boilerplate code in this class.